src/rgw/rgw_op.cc

```diff
@@ -1897,10 +1897,14 @@ void RGWPutMetadata::execute()
   /* no need to track object versioning, need it for bucket's data only */
   RGWObjVersionTracker *ptracker = (s->object ? NULL : &s->bucket_info.objv_tracker);
 
-  /* check if obj exists, read orig attrs */
-  ret = get_obj_attrs(store, s, obj, orig_attrs, NULL, ptracker);
-  if (ret < 0)
-    return;
+  if (s->object) {
+    /* check if obj exists, read orig attrs */
+    ret = get_obj_attrs(store, s, obj, orig_attrs, NULL, ptracker);
+    if (ret < 0)
+      return;
+  } else {
+    orig_attrs = s->bucket_attrs;
```

Comment:
it doesn't look like the existing meta attrs should be deleted according to the swift docs, but it looks like that's what this does with rmattrs.

http://docs.openstack.org/api/openstack-object-storage/1.0/content/POST_updateContainerMeta__v1__account___container__storage_container_services.html